gordo_components/cli/cli.py

```diff
@@ -181,62 +189,71 @@ def build(
-                name, model_config, data_config, metadata, evaluation_config
-            )
-
-    else:
-        model_location = provide_saved_model(
-            name,
-            model_config,
-            data_config,
-            metadata,
-            output_dir,
-            model_register_dir,
-            evaluation_config=evaluation_config,
-        )
-        metadata = load_metadata(model_location)
-
-    # If the model is cached but without CV scores then we force a rebuild. We do this
-    # by deleting the entry in the cache and then rerun `provide_saved_model`
-    # (leaving the old model laying around)
-    if print_cv_scores:
-        retrieved_metadata = metadata
-        all_scores = get_all_score_strings(retrieved_metadata)
-        if not all_scores:
-            logger.warning(
-                "Found that loaded model does not have cross validation values "
-                "even though we were asked to print them, clearing cache and "
-                "rebuilding model"
-            )
-
             model_location = provide_saved_model(
                 name,
                 model_config,
                 data_config,
                 metadata,
                 output_dir,
                 model_register_dir,
-                replace_cache=True,
                 evaluation_config=evaluation_config,
             )
-            saved_metadata = load_metadata(model_location)
-            all_scores = get_all_score_strings(saved_metadata)
-
-        for score in all_scores:
-            print(score)
+            metadata = load_metadata(model_location)
+
+        # If the model is cached but without CV scores then we force a rebuild. We do this
+        # by deleting the entry in the cache and then rerun `provide_saved_model`
+        # (leaving the old model laying around)
+        if print_cv_scores:
+            retrieved_metadata = metadata
+            all_scores = get_all_score_strings(retrieved_metadata)
+            if not all_scores:
+                logger.warning(
+                    "Found that loaded model does not have cross validation values "
+                    "even though we were asked to print them, clearing cache and "
+                    "rebuilding model"
+                )
+
+                model_location = provide_saved_model(
+                    name,
+                    model_config,
+                    data_config,
+                    metadata,
+                    output_dir,
+                    model_register_dir,
+                    replace_cache=True,
+                    evaluation_config=evaluation_config,
+                )
+                saved_metadata = load_metadata(model_location)
+                all_scores = get_all_score_strings(saved_metadata)
+
+            for score in all_scores:
+                print(score)
+    except Exception as e:
+        exit_code = EXCEPTION_TO_EXITCODE.get(e.__class__, 1)
+        traceback.print_exc()
+        sys.exit(exit_code)
```

Comment:
Small thing: add an `else` ?

Reply:
Containing what? The return?

Reply:
yespi. Just to emphasize that two possible outcomes can happen in in this `try` block. 
I guess this is a style choice if you prefer implicit or explicit forms similar to:
```python
if foo:
    return foo
return bar
```
or 
```python
if foo:
    return foo
else:
    return bar
```

Reply:
It's a free country, though.. :norway: